srlearn/base.py

```diff
@@ -186,18 +215,20 @@ def from_json(self, file_name):
             self._dotfiles = params[4]
         except IndexError:
             self._dotfiles = None
-            logging.warning("Did not find dotfiles during load, srlearn.plotting may not work.")
+            logging.warning(
+                "Did not find dotfiles during load, srlearn.plotting may not work."
+            )
 
         _bkg = Background()
-        _bkg.__dict__ = _model_parameters['background']
+        _bkg.__dict__ = _model_parameters["background"]
 
-        self.__init__(
-            background=_bkg,
-            target=_model_parameters['target'],
-            n_estimators=_model_parameters['n_estimators'],
-            node_size=_model_parameters['node_size'],
-            max_tree_depth=_model_parameters["max_tree_depth"],
-        )
+        _attributes = {"background": _bkg}
+        for key in BaseBoostedRelationalModel().__dict__.keys() - {"background"}:
+            _attributes[key] = _model_parameters.get(
+                key,
+                BaseBoostedRelationalModel().__dict__[key],
+            )
+        self.__init__(**_attributes)
```

Comment:
This might warrant a code comment:

1. Loop over keys in `BaseBoostedRelationalModel().__dict__.keys() - {"background"}`: i.e. all class attributes except background --- which has to be handled as a special case
2. Update a dictionary mapping parameters to values from loaded json
3. *Otherwise* (if there is a key that exists in the future and is not present in the json) use `.get(key, [,default]` default argument to set the default value of the current version.
4. Initialize the current object by overloading the created dictionary